nvflare/fuel/f3/cellnet/cell.py

```diff
@@ -761,13 +765,23 @@ def stop(self):
         if not self.running:
             return
 
+        # notify peers that I am gone
+        with self.agent_lock:
+            if self.agents:
+                targets = [peer_name for peer_name in self.agents.keys()]
+                self.logger.debug(f"broadcasting goodbye to {targets}")
+                self.broadcast_request(
```

Comment:
That's okay. This is just a "best effort" to notify the peer that I'm gone so that the peer can clear up its peer list. If the peer isn't available to process this message, it still has other opportunities to clean up: when the connection is gone, it will get notified, though there is no guarantee of timing when this happens. In a later version, we need to add a heartbeat mechanism at F3 layer to detect stale connections more quickly in case the connection event is delayed.